index.js

```diff
@@ -0,0 +1,119 @@
+const express = require("express");
+const app = express();
+const fs = require("fs");
+app.use(express.json());
+
+let data;
+try {
+  data = JSON.parse(fs.readFileSync(`${__dirname}/data.json`));
+  console.log("The required data", data);
+} catch (error) {
+  console.log("Error fetched", error.message);
+}
+
+// GET
+app.get("/info/:id", (request, response) => {
+  const { id } = request.params;
+
+  const item = data.find((user) => user.id === Number(id));
+  if (item) {
+    return response.status(200).json({
+      item,
+    });
+  }
+});
+
+// POST
+app.post("/info", (request, response) => {
+  const newId = data[data.length - 1].id + 1;
+  const newData = Object.assign({ id: newId }, request.body);
+  data.push(newData);
+  fs.writeFile(`${__dirname}/data.json`, JSON.stringify(data), (err) => {
+    response.status(200).json({
+      status: "success",
+      data: {
+        newData,
+      },
+    });
+  });
+});
+
+// PUT
+app.put("/info/:id", (request, response) => {
+  const { id } = request.params;
+  const updatedData = request.body;
```

Comment:
I suggest checking if the body has something so you don't allow users to edit a certain item using an empty object leaving it with the item with only the id.